File: lib/ts/src/internal/types/VInteger.ts
import { ValidationFailure } from '../validation/ValidationFailure';
import { VTypeDeclaration } from './VTypeDeclaration';
import { VType } from './VType';
import { validateInteger } from '../validation/ValidateInteger';
import { generateRandomInteger } from '../generation/GenerateRandomInteger';
import { GenerateContext } from '../generation/GenerateContext';
import { ValidateContext } from '../validation/ValidateContext';

export const integerName: string = 'Integer';

export class VInteger extends VType {
    getTypeParameterCount(): number {
        return 0;
    }

    validate(value: any, typeParameters: VTypeDeclaration[], ctx: ValidateContext): ValidationFailure[] {
        return validateInteger(value);
    }

    generateRandomValue(
        blueprintValue: any,
        useBlueprintValue: boolean,
        typeParameters: VTypeDeclaration[],
        ctx: GenerateContext,
    ): any {
        return generateRandomInteger(blueprintValue, useBlueprintValue, ctx);
    }

    getName(): string {
        return integerName;
    }
}

File: lib/java/src/main/java/io/github/telepact/internal/types/VInteger.java
package io.github.telepact.internal.types;

import static io.github.telepact.internal.generation.GenerateRandomInteger.generateRandomInteger;
import static io.github.telepact.internal.validation.ValidateInteger.validateInteger;

import java.util.List;

import io.github.telepact.internal.generation.GenerateContext;
import io.github.telepact.internal.validation.ValidateContext;
import io.github.telepact.internal.validation.ValidationFailure;

public class VInteger implements VType {
    public static final String _INTEGER_NAME = "Integer";

    @Override
    public int getTypeParameterCount() {
        return 0;
    }

    @Override
    public List<ValidationFailure> validate(Object value, List<VTypeDeclaration> typeParameters, ValidateContext ctx) {
        return validateInteger(value);
    }

    @Override
    public Object generateRandomValue(Object blueprintValue, boolean useBlueprintValue,
            List<VTypeDeclaration> typeParameters, GenerateContext ctx) {
        return generateRandomInteger(blueprintValue, useBlueprintValue, ctx);
    }

    @Override
    public String getName() {
        return _INTEGER_NAME;
    }
}
